### DISCUSSION TITLE
AccountGoDev Slack Channel

### DISCUSSION THREAD

**Discussion by mvperez** [AccountGo/accountgo#176] [Team Posts]
Hi guys,
We have a slack channel "accountgodev". If you wish to be added in that workspace please do let me know your github/email address so I can send you invite.
Thanks

**Comment by mvperez** [comment]
Hi @vivencio2 , have you had the chance to update the readme.md? I think it would be very helpful if we write down your update before we forget it.
Thanks

**Comment by vivencio2** [comment]
Yes @marvinvperez currently doing it. I'm just making all global dependency installations be moved in the initial setup process prior to project restore and build so it will be seamless and error free. I'm also trying to explicitly detail it as much as I can.

**Comment by mvperez** [comment]
Hi @vivencio2, your PR is now merged. About the user, roles, and permission management, I am thinking if this was a good start to implement modularity. I already started it on "Infrastructure" project - checkout the branch `modules` for the latest code, some of it already in the master. It has sample module project in test folder. 
For sample patterns, I am using these resources as inspirations

https://github.com/thiennn/trymodular
https://github.com/natemcmaster/DotNetCorePlugins
https://github.com/nopSolutions/nopCommerce/tree/develop/src/Libraries/Nop.Core/Plugins

What I'd like to suggest -
1. Implement user, roles, permissions on the current design / architecture.
2. Move #1 to a module

The challenging part is to make the module really independent, which I don't know yet how to do. What I mean is, because the web front-end is separate from api back-end, then it's impossible to make it totally independent module because we still have to touch the api for the database actions. But I think this is okay, I mean nothing we can do about it.

Many times you can see plugins and modules from other systems seems the same. But to us, by definition we will make it a module if it seems like a subsystem e.g. contains views, controllers, models etc. We will call plugins if its too small for a module - no controllers, views, and models. Purely logic and calculation.

Something like in api, we can add plugins. In frontend, we can modules.

I'm happy to get ideas and feedback.

**Comment by vivencio2** [comment]
I'll work on this this weekend and will add to the discussion.

**Comment by vivencio2** [comment]
just to update, I already started to work on the user, roles and permissions last sunday, I'll probably have something to checkin by thursday :)

**Comment by mvperez** [comment]
Hi @vivencio2 , how are you?

**Comment by vivencio2** [comment]
hey @marvinvperez i'm good was just got busy with some office stuff. I will get back on what I left that I supposed to checkin last thursday. I'll checkin soon.

**Comment by mvperez** [comment]
Thank @vivencio2! Maybe the next step is to enable third party authentication e.g. Facebook, Google.
I would eagerly publish the software to marketplace such as AppDirect (https://www.appdirect.com) and IngramMicro (https://www.ingrammicro.com)

**Comment by mvperez** [comment]
Hope everything is fine @vivencio2

**Comment by vivencio2** [comment]
hey @marvinvperez I was just busy with office projects but just deployed so I can continue on the committed task I have here :)

**Comment by mvperez** [comment]
Hi @vivencio2 , no worries. I'm close to merging to master my initial work on modular, but I think I will wait first on your first major PR.

**Comment by mvperez** [comment]
I merge to master the initial work on modular design. Theres a SampleModule project. The web front-end will load in startup, and can be access from http://localhost:8000/samplemodule. Still lots of work but a good start. :)

**Comment by mvperez** [comment]
Hi @vivencio2 , how are you?

**Comment by vivencio2** [comment]
hi @marvinvperez still busy but I will be back here by December 20th onwards. Sorry.
